matchzoo/contrib/models/diin.py

```diff
@@ -0,0 +1,310 @@
+                         desc="The filter size of each convolution layer in "
+                              "dense block."))
+        params.add(Param(name='transition_scale_down_ratio', value=0.5,
+                         desc="The channel scale down ratio of the "
+                              "convolution layer in transition block."))
+        return params
+
+    @classmethod
+    def get_default_preprocessor(cls):
+        """:return: Default preprocessor."""
+        return preprocessors.DIINPreprocessor()
+
+    def guess_and_fill_missing_params(self, verbose: int = 1):
+        """
+        Guess and fill missing parameters in :attr:'params'.
+
+        Use this method to automatically fill-in hyper parameters.
+        This involves some guessing so the parameter it fills could be
+        wrong. For example, the default task is 'Ranking', and if we do not
+        set it to 'Classification' manually for data packs prepared for
+        classification, then the shape of the model output and the data will
+        mismatch.
+
+        :param verbose: Verbosity.
+        """
+        self._params.get('input_shapes').set_default([(32,),
+                                                      (32,),
+                                                      (32, 16),
+                                                      (32, 16),
+                                                      (32,),
+                                                      (32,)], verbose)
+        super().guess_and_fill_missing_params(verbose)
+
+    def _make_inputs(self) -> list:
+        text_left = keras.layers.Input(
+            name='text_left',
+            shape=self._params['input_shapes'][0]
+        )
+        text_right = keras.layers.Input(
+            name='text_right',
+            shape=self._params['input_shapes'][1]
+        )
+        char_left = keras.layers.Input(
+            name='char_left',
+            shape=self._params['input_shapes'][2]
+        )
+        char_right = keras.layers.Input(
+            name='char_right',
+            shape=self._params['input_shapes'][3]
+        )
+        match_left = keras.layers.Input(
+            name='match_left',
+            shape=self._params['input_shapes'][4]
+        )
+        match_right = keras.layers.Input(
+            name='match_right',
+            shape=self._params['input_shapes'][5]
+        )
+        return [text_left, text_right,
+                char_left, char_right,
+                match_left, match_right]
+
+    def build(self):
+        """Build model structure."""
+
+        # Scalar dimensions referenced here:
+        #   B = batch size (number of sequences)
+        #   D = word embedding size
+        #   L = 'input_left' sequence length
+        #   R = 'input_right' sequence length
+        #   C = fixed word length
+
+        # Left text and right text.
+        # shape = [B, L]
+        # shape = [B, R]
+        text_left, text_right = self._make_inputs()[0:2]
```

Comment:
Shouldn't call this multiple times. Do a `inputs = self._make_inputs()` then slice `inputs`.